pypeit/spectrographs/p200_dbsp.py

```diff
@@ -0,0 +1,432 @@
+            frametype = headarr[0]['IMGTYPE']
+            if am is None and frametype in ['bias', 'flat', 'cal']:
+                am = '1.000'
+            elif headarr[0].get('RA', default=None) and headarr[0].get('DEC', default=None):
+                ra = headarr[0]['RA']
+                dec = headarr[0]['DEC']
+                altaz = SkyCoord(ra, dec, unit=(u.hour, u.deg)).transform_to(AltAz(obstime=headarr[0]['UTSHUT'], location=loc))
+                am = altaz.secz
+            return am
+        else:
+            return None
+            #msgs.error("Not ready for this compound meta")
+
+    def pypeit_file_keys(self):
+        pypeit_keys = super(P200DBSPSpectrograph, self).pypeit_file_keys()
+        pypeit_keys += ['slitwid']
+        return pypeit_keys
+
+
+class P200DBSPBlueSpectrograph(P200DBSPSpectrograph):
+    """
+    Child to handle P200/DBSP blue specific code
+    """
+    def __init__(self):
+        # Get it started
+        super(P200DBSPBlueSpectrograph, self).__init__()
+        self.spectrograph = 'p200_dbsp_blue'
+        self.camera = 'DBSPb'
+    
+    def compound_meta(self, headarr, meta_key):
+        retval = super(P200DBSPBlueSpectrograph, self).compound_meta(headarr, meta_key)
+        if retval is not None:
+            return retval
+        if meta_key == 'binning':
+            binspatial, binspec = headarr[0]['CCDSUM'].split(' ')
+            return parse.binning2string(binspec, binspatial)
+        else:
+            msgs.error("Not ready for this compound meta")
+
+    def get_detector_par(self, hdu, det):
+        """
+        Return a DectectorContainer for the current image
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`):
+                HDUList of the image of interest.
+                Ought to be the raw file, or else..
+            det (int):
+
+        Returns:
+            :class:`pypeit.images.detector_container.DetectorContainer`:
+
+        """
+        # Binning
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning')  # Could this be detector dependent??
+
+        # Detector 1
+        detector_dict = dict(
+            binning         = binning,
+            det             = 1,
+            dataext         = 0,
+            specaxis        = 0,
+            specflip        = True,
+            spatflip        = False, # check
+            platescale      = 0.389,
+            darkcurr        = 0.0,
+            saturation      = 65000.,
+            nonlinear       = 62./65.,
+            mincounts       = -1e10, # cross-check
+            numamplifiers   = 1,
+            gain            = np.atleast_1d(0.72),
+            ronoise         = np.atleast_1d(2.5),
+            datasec         = np.atleast_1d('[1:2835,1:410]'),       # TODO: from datamodel, should just be able to point this to DSEC1/BSEC1
+            oscansec        = np.atleast_1d('[1:2835,411:460]')      # but that just doesn't work, so we will hardcode.
```

Comment:
If you can get this info from the header, you should.
See `shane_kast_red` for an example.